### PR TITLE
Model support for LLama and Qwen

### LINKED ISSUE #14: Support more models
Gemma models require non-standard variants of attention like softmax logit capping. This is currently not supported in yalis.

### PR DESCRIPTION
**Pull request by ishanrev**
New models configs have been added to the external/config file. 

Qwen 2.5 and 3 models can now be downloaded. Qwen 3 without MOE support. (Still working on it).

New download file (taken from litgt)

Note: 
So all tests are passing, only thing is during the tests the dist.barrier fails ( CUDA couldn't allocate more space), not sure what the issue is, but normal runs with tensor parallelism and everything works.

### CODE REVIEW

File: yalis/external/model.py [hunk 1]
```diff
@@ -523,6 +526,16 @@ def __init__(self, config: Config, block_idx: int) -> None:
             config.sliding_window_size is not None
             and block_idx % config.sliding_window_layer_placing == 0
         )
+        
+        if config.norm_qk:
+            norm_q_size = config.n_head * config.head_size if config.norm_qk_type == "olmo2" else config.head_size
```
Comment (prajwal1210) [inline yalis/external/model.py]: Has olmo been tested?
Particularly, want to know if all code paths added have been tested or not?

File: yalis/external/config.py [hunk 2]
```diff
@@ -2331,12 +2575,83 @@ def norm_class(self) -> Type:
...
+        intermediate_size=27648,
+        norm_eps=1e-5,
+        rope_base=1000000,
+    ),
     # https://huggingface.co/Qwen/QwQ-32B-Preview/blob/main/config.json
     dict(
         name="QwQ-32B-Preview",
         hf_config=dict(org="Qwen", name="QwQ-32B-Preview"),
-        block_size=131072,
+        block_size=32768,
```
[the remark this answers is not in the document]
Comment (ishanrev) [reply yalis/external/config.py:2737]: Just sticking to what litgptss latest config file has in terms of specifications for these models

File: yalis/external/download.py [hunk 3]
```diff
@@ -100,56 +99,20 @@ def download_from_hub(
...
     with gated_repo_catcher(repo_id, access_token):
         info = repo_info(repo_id, token=access_token)
     filenames = [f.rfilename for f in info.siblings]
-    bins = list(
-        filter_repo_objects(items=filenames, allow_patterns=["*.bin*"])
-    )
-    safetensors = list(
-        filter_repo_objects(items=filenames, allow_patterns=["*.safetensors*"])
-    )
+    bins = list(filter_repo_objects(items=filenames, allow_patterns=["*model*.bin*"]))
```
[the remark this answers is not in the document]
Comment (ishanrev) [reply yalis/external/download.py]: Latest download file from litgpt uses this therefore Not making this change

File: yalis/external/model.py [hunk 4]
```diff
@@ -523,6 +526,16 @@ def __init__(self, config: Config, block_idx: int) -> None:
             config.sliding_window_size is not None
             and block_idx % config.sliding_window_layer_placing == 0
```
[the remark this answers is not in the document]
Comment (ishanrev) [reply yalis/external/model.py:525]: Not a change I made

**Review by prajwal1210 — APPROVED** [review]
